### PR TITLE
#56 Implementing list view for Scheme Type

### PR DESCRIPTION
**Pull request by bkuzmic**
For #56:

- Implementing List view for Scheme Types
- Leaving puzzles for further development

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#337](https://github.com/timelesslounge/timelessis/issues/337) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#337](https://github.com/timelesslounge/timelessis/issues/337) is assigned to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/76676a4c-6ff4-4560-9387-3f43962783ff); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: tests/integration/it_scheme_types_test.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from http import HTTPStatus
+from timeless.schemetypes.models import SchemeType
```
Comment (vladarefiev) [inline tests/integration/it_scheme_types_test.py]: @bkuzmic should be blank line before it. PEP8

File: tests/integration/it_scheme_types_test.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+
+def test_list(client, db_session):
+    """ Test getting list of Scheme types """
+    db_session.add(SchemeType(description="Test scheme type",
+                              default_value="1", value_type="Integer"))
+    db_session.commit()
+    types = client.get("/schemetypes/")
```
Comment (vladarefiev) [inline tests/integration/it_scheme_types_test.py]: @bkuzmic does `url_for` work here? if yes, use it pls

File: timeless/schemetypes/views.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+@todo #56:30min Continue implementing Create, Edit and Delete views once
+ generic views from #173 are implemented. Code templates are already provided
+ below, so uncomment and modify - create templates and write ITs to verify
+ behaviour.
+@todo #56:30min Continue implementing List to enable sorting and filtering
+ for every column. For this, probably there will be a need to create a new
+ generic view that all other List views will extend. This generic view should
+ use GenericFilter implemented in #317.
+"""
+from flask import Blueprint
```
Comment (vladarefiev) [inline timeless/schemetypes/views.py]: @bkuzmic just cosmetic
```
from flask import Blueprint

from timeless.views import ListView
from timeless.schemetypes.models import SchemeType


bp = Blueprint("scheme_type", __name__, url_prefix="/schemetypes")
```

File: timeless/templates/schemetypes/list.html [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+{% block header %}
+    <h1>{% block title %}Scheme Types management - Main{% endblock %}</h1>
+{% endblock %}
+
+{% block content %}
+    {% for type in object_list %}
+        <article class="schemetype">
+        <header>
+            <div>
+            <h1>{{ type['description'] }}</h1>
```
Comment (vladarefiev) [inline timeless/templates/schemetypes/list.html]: @bkuzmic one tab here

File: timeless/templates/schemetypes/list.html [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{% extends 'base.html' %}
+{% cache 600 %}
+{% block header %}
+    <h1>{% block title %}Scheme Types management - Main{% endblock %}</h1>
+{% endblock %}
+
+{% block content %}
+    {% for type in object_list %}
+        <article class="schemetype">
+        <header>
```
Comment (vladarefiev) [inline timeless/templates/schemetypes/list.html]: @bkuzmic one tap here

**Comment by vladarefiev** [comment]
@bkuzmic Looks really good to me

**Comment by bkuzmic** [comment]
@vladarefiev Thanks for the comments. Please take a look now.

**Comment by vladarefiev** [comment]
@rultor good to merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 23 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job [#337](https://github.com/timelesslounge/timelessis/issues/337) is not in the agenda of @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), can't inspect

**Comment by 0crat** [comment]
The job [#337](https://github.com/timelesslounge/timelessis/issues/337) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/timelesslounge/timelessis/issues/337#issuecomment-466770176))

@ypshenychka This is what I know about this job in [CF9LG1YCA](https://www.0crat.com/p/CF9LG1YCA), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#337](https://github.com/timelesslounge/timelessis/issues/337) is not in scope
  * The job is waiting quality review verdict by @ypshenychka[/z](https://www.0crat.com/u/ypshenychka) for 3days
  * @vladarefiev[/z](https://www.0crat.com/u/vladarefiev) will receive 20 minutes after quality review
  * Job [footprint](https://www.0crat.com/footprint/CF9LG1YCA?q=%7Bjob%3A%27gh%3Atimelesslounge%2Ftimelessis%23337%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by ypshenychka** [comment]
@vladarefiev  According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only one major issue was found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by vladarefiev** [comment]
@ypshenychka confirm

**Comment by ypshenychka** [comment]
@vladarefiev thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +20 point(s) just awarded to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**